Common/Core/RecoDecay.h

```diff
@@ -678,22 +671,8 @@ class RecoDecay
     //printf("Stage %d: %d (PDG %d) -> %d-%d\n", stage, index, PDGParticle, indexDaughterFirst, indexDaughterLast);
     // Call itself to get daughters of daughters recursively.
     stage++;
-    // Get daughters of the first daughter.
-    if (indexDaughterFirst > -1) {
-      getDaughters(particlesMC, particlesMC.iteratorAt(indexDaughterFirst), list, arrPDGFinal, depthMax, stage);
-    }
-    // Get daughters of the daughters in between if any.
-    // Daughter indices are supposed to be consecutive and in increasing order.
-    // Reverse order means two daughters.
-    if (indexDaughterFirst > -1 && indexDaughterLast > -1) {
-      for (auto iD = indexDaughterFirst + 1; iD < indexDaughterLast; ++iD) {
-        getDaughters(particlesMC, particlesMC.iteratorAt(iD), list, arrPDGFinal, depthMax, stage);
-      }
-    }
-    // Get daughters of the last daughter if different from the first one.
-    // Same indices indicate a single daughter.
-    if (indexDaughterLast > -1 && indexDaughterLast != indexDaughterFirst) {
-      getDaughters(particlesMC, particlesMC.iteratorAt(indexDaughterLast), list, arrPDGFinal, depthMax, stage);
+    for (auto& dau : particle.template daughters_as<typename std::decay_t<T>::parent_t>()) {
+      getDaughters(dau, list, arrPDGFinal, depthMax, stage);
     }
```

Comment:
@vkucera when this piece of code is executed I get the following error:
`[ERROR] Exception caught: Unable to find column with label fIndexMcCollisions`
even if I checked and the column `fIndexMcCollisions` is present in the `O2mccollisionlabel` table in the `AO2D` file that I am analysing. 
I wrote this piece of code following what was suggested by @aalkin for 
`particleMother = particleMother.template mothers_first_as<typename std::decay_t<T>::parent_t>();` 
however I am not sure that it works also in this case. I also tried by replacing `typename std::decay_t<T>::parent_t` with simply `aod::McParticles` but I get the same error. @aalkin @jgrosseo do you have any suggestion? Thanks!

Reply:
@fgrosa this particular exception is raised when constructing a table. When the slice index is unset it is supposed to return empty table. It may happen that this actually creates an empty table with no schema and thus the construction fails. For now you can protect this piece of code with check for `has_daughters()`.

Reply:
Hi @aalkin thanks! Unfortunately even if I put the protection:
```cpp
    if (particle.has_daughters()) {
      for (auto& dau : particle.template daughters_as<typename std::decay_t<T>::parent_t>()) {
        getDaughters(dau, list, arrPDGFinal, depthMax, stage);
      }
    }
```
it keeps crashing with the same error. Do you have any other idea by chance?

Reply:
I will have to debug this, I assume the line `auto dauTable = particle.template daughters_as<typename std::decay_t<T>::parent_t>();` would crash as well, could you check?

Reply:
Checking for `has_daughters()` has no effect here because this line is reached only if `has_daughters()` returned `true`.

Reply:
@aalkin indeed it crashes as well. @vkucera right because if `isFinal` is `true` and `has_daughters()` is false it goes through the first `if` statement, but the it returns anyway before this part of the code. 
I think that one possible solution would be to loop over indices in this case instead of getting the sliced table of daughters, but it would require to pass again the MC particle table as argument. However I have no better ideas at the moment. What do you think?

Reply:
@aalkin Would this work?
```c++
  template <std::size_t N, typename T>
  static void getDaughters(const typename T::iterator& particle,
    ...
    for (auto& dau : particle.daughters_as<T>()) {
```

Reply:
@vkucera I doubt this will be able to deduce type automatically from a function call, but it is better to check explicitly.

Reply:
@vkucera @aalkin I already tried this, but the compilation fails with the following error: `couldn't deduce template parameter 'T'`